### PR TITLE
a TIL about vim plugin managers

### PR DESCRIPTION
**Pull request by mehulkar**
I don't know 100% if the thing I have learned is correct/accurate.

Exercising [Cunningham's law](https://meta.wikimedia.org/wiki/Cunningham%27s_Law) here.

### CODE REVIEW

**Comment by scanf** [comment]
+1

File: vim/plugin-managers.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Plugin Managers
+
+Today I learned that vim has two main plugin managers:
```
Comment (mike-burns) [inline vim/plugin-managers.md]: There's also the classic https://packages.debian.org/sid/vim-addon-manager from 2007.

**Comment by ghiden** [comment]
Vim-plug is kind of new but very nice.
https://github.com/junegunn/vim-plug
I recently switched from Pathogen to vim-plug and can't go back.

**Comment by mehulkar** [comment]
@mike-burns @ghiden want me to change anything in the PR?

**Comment by ghiden** [comment]
@mehulkar If you can mention vim-plug, that would be great.

**Comment by ianks** [comment]
:+1:  for vim-plug. I would also mention NeoBundle as well. Both of those offer more functionality and are more well-maintained than Vundle.

File: vim/plugin-managers.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+1. Tim Pope<sup>1</sup>'s [Pathogen][1]
+1. gmarik's [Vundle][2]
```
Comment (dleve123) [inline vim/plugin-managers.md]: Pretty sure this should be "**2**. gmarik's [Vundle][2]". Interestingly, the raw file renders correctly...

Comment (mehulkar) [inline vim/plugin-managers.md]: Markdown doesn't care about what the numbers are. It's just converting numeric values to <ol><li> elements and HTML is doing the ordering. Hey! That could be your TIL :smile:

Comment (dleve123) [inline vim/plugin-managers.md]: Very cool - learn something new erryday. I wonder if it's a best practice though? 

[The list section](http://daringfireball.net/projects/markdown/syntax#list) of the Daring Fireball guide seems to suggest that it is a "lazy" practice. However, I also definitely see how allows one to manage large and high churn lists easily without the fear of breaking list ordering.

Some food for thought!

Comment (mehulkar) [inline vim/plugin-managers.md]: [laziness is good](http://threevirtues.com/) :smile:

**Comment by mehulkar** [comment]
@ianks @ghiden @mike-burns not sure who has commit rights, but I updated the PR mentioning the other managers as well!

File: vim/plugin-managers.md (hunk 2 above)
Comment (dleve123) [inline vim/plugin-managers.md]: This is true! :+1:

File: vim/plugin-managers.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+--
+
+**Footnotes**
+
+1. Tim Pope is kind of a vim legend, from what I gather.
+1. I guess that's a weird time paradox. Here, "Today" refers to the original TIL day.
+
+[1]: https://github.com/tpope/vim-pathogen
+[2]: https://github.com/gmarik/Vundle.vim
+[3]: https://packages.debian.org/sid/vim-addon-manager
```
Comment (Keithbsmiley) [inline vim/plugin-managers.md]: I think https://github.com/MarcWeber/vim-addon-manager would be a better URL

File: vim/plugin-managers.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+The biggest difference is that with Pathogen, you have to clone
+and install plugins on the filesystem manually. With Vundle,
+you add directives to your `~/.vimrc` and run a command
+to install/update the plugins, thus abstracting vim's away
+how vim's plugin system works.
+
+Both plugin managers are also plugins. Yeah.
+
+**TDATIL (The Day After Today I Learned)**
```
Comment (Keithbsmiley) [inline vim/plugin-managers.md]: I don't think this section is necessary, I think it would make more sense to put all of these together. Up to you though.

**Comment by thorncp** [comment]
Is the expectation going forward that all new package managers will be added to this list? It seems to me like it has increased beyond the scope of a Today I Learned.

**Comment by JoelQ** [comment]
@thorncp Agreed. Thoughts on just keeping the original TIL and say that Pathogen and Vundle are the two most popular plugin managers?

**Comment by Keithbsmiley** [comment]
I think, if that's the plan we should think about what we really want TIL to be. IMO if we go down that route, this information is already out of date. But I also don't think we should plan to accept pull requests each time a new fringe plugin manager is released.

**Comment by mehulkar** [comment]
TIL that TIL is pretty complicated. I mean...what really is today? What really is time!?

**Comment by JoelQ** [comment]
@Keithbsmiley TIL is closer much closer to our blog than it is to our guides. TILs won't be updated when they become obsolete, they are neither definitive nor exhaustive.

I'm fine with a TIL that is "here is the 80% solution for managing VIM plugins" . We don't need to cover all the options. Feel free to make the case that Vundle/Pathogen is not the majority/preferred plugin management solution in the VIM community.

**Comment by mehulkar** [comment]
> Feel free to make the case that Vundle/Pathogen is not the majority/preferred plugin management solution in the VIM community.

Does the "I" in TIL supposed to indicate the author or community consensus?

**Comment by JoelQ** [comment]
> Is the "I" in TIL supposed to indicate the author or community consensus?

"I" is the author. 

Your learning however was about the community's preferred installation solution:

> Today I learned that vim has two main plugin managers:

**Comment by keith** [comment]
Can we close or merge this one?

**Comment by mehulkar** [comment]
closing

**Pull request closed without merging**